### ISSUE TITLE
event for changes in files.length?

### ISSUE THREAD

**Issue by davidmaxwaterman** [vaadin/vaadin-upload#183]
I have vaadin-upload in a paper-dialog. When files are selected, I'd like to re-fit the dialog by calling this.$.dialog.fit() when files are added/removed from the dialog (which result in the height of the dialog changing.

Please consider adding such an event.

**Comment by davidmaxwaterman** [comment]
Actually, I found it best to tie this to the dom-change event for the template (files.length changes before the dom changes, so fit()ing doesn't work).

**Comment by Saulis** [comment]
Hi!

If the `dom-change` event indeed bubbles through, that's currently probably your best option.

We could consider firing a custom event like `iron-resize` or something to notify more explicitly that the element height has changed.

**Comment by davidmaxwaterman** [comment]
yeah, I did have a 'flash of a thought' about dom-change bubbling, but dismissed it. I'll give it a try. Thanks.

**Comment by davidmaxwaterman** [comment]
FWIW, using `on-dom-change="_handleDomChange` *does* work, so I'm good.

However, I kind of feel like it's a bit 'ugly' and vaadin-upload should have its own event for this. Actually, without looking at the internals, a user of this element might assume that the event relates to the vaadin-upload element itself, rather than a dom-repeat template within it...sure, it doesn't take much thinking to see what it relates to, but......anyway, up to you :) I'd at least recommending adding that to the docs.
